packages/framework/src/types/event.types.ts

```diff
@@ -69,7 +59,15 @@ export type EventTriggerParams<T_Payload = EventPayload> = {
       [stepId: string]: Record<string, unknown>;
     };
   };
-};
+} & ConditionalPartial<
+  {
+    /**
+     * Payload to trigger the workflow with
+     */
+    payload: T_Payload;
+  },
+  PickRequiredKeys<T_Payload> extends never ? true : false
+>;
```

Comment:
Using a new utility type which marks the first generic argument as partial when the second generic argument is `true`. Combined with `PickRequiredKeys` which returns `never` when there are no required keys, we can make `payload` optional when there are no required keys present in the `payload`.